arangod/VocBase/LogicalView.h

```diff
@@ -163,10 +163,26 @@ class DBServerLogicalView : public LogicalView {
     bool isNew
   );
 
+  //////////////////////////////////////////////////////////////////////////////
+  /// @brief fill and return a jSON description of a View object implementation
+  //////////////////////////////////////////////////////////////////////////////
+  virtual void getPropertiesVPack(
+    velocypack::Builder& builder,
+    bool forPersistence
+  ) const = 0;
+
+  ///////////////////////////////////////////////////////////////////////////////
+  /// @brief called when a view's properties are updated (i.e. delta-modified)
+  ///////////////////////////////////////////////////////////////////////////////
+  virtual arangodb::Result updateProperties(
```

Comment:
why do we need this function?

Reply:
it is called when a view's properties are updated (i.e. delta-modified)